### PR TITLE
Feature: migrate Resource object to new TF SDK, breaking changes for protocols section

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves migrating provider to new SDK

Tests passes [UPD]: https://github.com/vmanilo/terraform-provider-twingate/actions/runs/7292375623

## Changes
- migrated resource object to new TF SDK
- breaking changes for `protocols`: blocks -> attributes
- breaking changes for `is_browser_shortcut_enabled` = false by default
- breaking changes for `is_visible` = true by default

### CODE REVIEW

File: docs/resources/resource.md [hunk 1]
```diff
@@ -64,12 +64,13 @@ resource "twingate_resource" "resource" {
...
 - `alias` (String) Set a DNS alias address for the Resource. Must be a DNS-valid name string.
 - `is_authoritative` (Boolean) Determines whether assignments in the access block will override any existing assignments. Default is `true`. If set to `false`, assignments made outside of Terraform will be ignored.
 - `is_browser_shortcut_enabled` (Boolean) Controls whether an "Open in Browser" shortcut will be shown for this Resource in the Twingate Client.
 - `is_visible` (Boolean) Controls whether this Resource will be visible in the main Resource list in the Twingate Client.
-- `protocols` (Block List, Max: 1) Restrict access to certain protocols and ports. By default or when this argument is not defined, there is no restriction, and all protocols and ports are allowed. (see [below for nested schema](#nestedblock--protocols))
+- `protocols` (Attributes) Restrict access to certain protocols and ports. By default or when this argument is not defined, there is no restriction, and all protocols and ports are allowed. (see [below for nested schema](#nestedatt--protocols))
```
Comment (twingate-blee) [inline docs/resources/resource.md:72]: Update the protocols "example usage" with the new format

File: twingate/internal/provider/resource/resource.go [hunk 2]
```diff
@@ -4,444 +4,966 @@ import (
...
+					upgradedState.Alias = priorState.Alias
+				}
+
+				if !priorState.SecurityPolicyID.IsNull() && priorState.SecurityPolicyID.ValueString() != "" {
+					upgradedState.SecurityPolicyID = priorState.SecurityPolicyID
+				}
+
+				resp.Diagnostics.Append(resp.State.Set(ctx, upgradedState)...)
+
+				resp.Diagnostics.AddWarning("Please upgrade protocols sections", "Follow this docs to update protocols from blocks to attributes")
```
Comment (vmanilo) [inline twingate/internal/provider/resource/resource.go:361]: we should add needed info to this warning msg

Comment (vmanilo) [reply twingate/internal/provider/resource/resource.go:367]: @twingate-blee @alexb-twingate what about this warning message, how should it look like?

File: docs/resources/resource.md [hunk 3]
```diff
@@ -70,13 +70,13 @@ resource "twingate_resource" "resource" {
...
 - `alias` (String) Set a DNS alias address for the Resource. Must be a DNS-valid name string.
 - `is_authoritative` (Boolean) Determines whether assignments in the access block will override any existing assignments. Default is `true`. If set to `false`, assignments made outside of Terraform will be ignored.
-- `is_browser_shortcut_enabled` (Boolean) Controls whether an "Open in Browser" shortcut will be shown for this Resource in the Twingate Client.
-- `is_visible` (Boolean) Controls whether this Resource will be visible in the main Resource list in the Twingate Client.
-- `protocols` (Block List, Max: 1) Restrict access to certain protocols and ports. By default or when this argument is not defined, there is no restriction, and all protocols and ports are allowed. (see [below for nested schema](#nestedblock--protocols))
-- `security_policy_id` (String) The ID of a `twingate_security_policy` to set as this Resource's Security Policy. Default is `Default Policy`
+- `is_browser_shortcut_enabled` (Boolean) Controls whether an "Open in Browser" shortcut will be shown for this Resource in the Twingate Client. Default is false.
```
[the remark this answers is not in the document]
Comment (vmanilo) [reply docs/resources/resource.md]: updated

File: docs/resources/resource.md [hunk 4]
```diff
@@ -70,13 +70,13 @@ resource "twingate_resource" "resource" {
...
+- `is_visible` (Boolean) Controls whether this Resource will be visible in the main Resource list in the Twingate Client. Default is true.
```
[the remark this answers is not in the document]
Comment (vmanilo) [reply docs/resources/resource.md]: updated

File: twingate/internal/provider/resource/resource.go (hunk 2 above)
Comment (twingate-blee) [reply twingate/internal/provider/resource/resource.go:367]: @alexb-twingate How about 
"Please update the protocols sections format from a block to an object", "There is a protocols object example in the twingate_resource documentation"

Comment (alexb-twingate) [reply twingate/internal/provider/resource/resource.go:367]: I like that. I'd change the longer description to "See the v1 to v2 migration guide in the Twingate Terraform Provider documentation" per what we talked about earlier today.

Comment (vmanilo) [reply twingate/internal/provider/resource/resource.go]: updated this message

**Comment by alexb-twingate** [comment]
@vmanilo - I took a pass at updating the migration guide. Are we also deprecating `is_admin` finally? We've had a deprecation notice for a while so I assumed we're finally removing that attribute.

**Review by romankor — APPROVED** [review]
GTM

**Pull request merged**